src/WebJobs.Script.WebHost/Security/KeyManagement/BlobStorageSecretsMigrationRepository.cs

```diff
@@ -0,0 +1,158 @@
+        }
+
+        public async Task<string[]> GetSecretSnapshots(ScriptSecretsType type, string functionName)
+        {
+            return await BlobStorageSecretsRepository.GetSecretSnapshots(type, functionName);
+        }
+
+        public void Dispose()
+        {
+            BlobStorageSecretsRepository.Dispose();
+        }
+
+        private async Task CopyKeysFromFileSystemToBlobStorage(BlobStorageSecretsRepository blobStorageSecretsRepository)
+        {
+            string migrateSentinelPath = Path.Combine(blobStorageSecretsRepository.SecretsSentinelFilePath, "migrate-sentinel.json");
+            if (File.Exists(migrateSentinelPath))
+            {
+                // Migration is already done
+                Log("Sential file is detected.");
+                return;
+            }
+
+            try
+            {
+                using (var stream = new FileStream(migrateSentinelPath, FileMode.CreateNew))
+                using (var writer = new StreamWriter(stream))
+                {
+                    //write file
+                    Log("Sential file created.");
+                }
+            }
+            catch (IOException)
+            {
+                Log("Sential file is already created by another instance.");
+                return;
+            }
+
+            string[] files = Directory.GetFiles(blobStorageSecretsRepository.SecretsSentinelFilePath.Replace("Sentinels", string.Empty));
+            if (await blobStorageSecretsRepository.BlobContainer.ExistsAsync())
+            {
+                BlobResultSegment resultSegment = await blobStorageSecretsRepository.BlobContainer.ListBlobsSegmentedAsync(blobStorageSecretsRepository.SecretsBlobPath + "/", null);
+
+                // Check for conflicts
+                if (resultSegment.Results.ToArray().Length > 0)
+                {
+                    Log("Conflict detected. Secrets container is not empty.");
+                    return;
+                }
+            }
+            else
+            {
+                await blobStorageSecretsRepository.BlobContainer.CreateIfNotExistsAsync();
+            }
+
+            if (files.Length > 0)
+            {
+                List<Task> copyTasks = new List<Task>();
+                foreach (string file in files)
+                {
+                    string blobName = Path.GetFileName(file);
+                    CloudBlockBlob cloudBlockBlob = blobStorageSecretsRepository.BlobContainer.GetBlockBlobReference(blobStorageSecretsRepository.SecretsBlobPath + "/" + blobName);
+                    Task copyTask = cloudBlockBlob.UploadFromFileAsync(file);
+                    copyTasks.Add(copyTask);
+                    Log("'{0}' was migrated.", cloudBlockBlob.StorageUri.PrimaryUri.AbsoluteUri.ToString());
+                }
+                await Task.WhenAll(copyTasks);
+            }
+            Log("Finished successfully.");
+        }
+
+        private void Log(string message, params object[] args)
```

Comment:
This method is not necessary and not the pattern used throughout the codebase. Suggest removing and replacing calls with `_logger?.LogTrace(..)`